core/src/socket.rs

```diff
@@ -14,15 +22,22 @@ use std::{
 
 pub type SocketHandle = Index;
 
+#[derive(Copy, Clone, Collect)]
+#[collect(no_drop)]
+enum SocketKind<'gc> {
+    Avm2(SocketObject<'gc>),
+    Avm1(Object<'gc>),
```

Comment:
It might be nice to import `Object` as `Avm1Object`, to avoid confusion in other parts of this file.